src/services/proposals.ts

```diff
@@ -30,3 +31,80 @@ export async function createProposalOnDb(mySql: ServerlessMysql, data: CreatePro
 
   return { proposalId };
 }
+
+export interface IProposalSqlRow {
+  proposal: string,
+  hashed_auth_password: string,
+  auth_password_salt: string,
+  version: number,
+  partial_tx: string,
+  signatures: string,
+  history: string,
+  created_at: string,
+  updated_at: string,
+}
+
+export interface IDbProposal {
+  id: string,
+  partialTx: string,
+  signatures: string,
+  timestamp: string,
+  version: number,
+  history: {partialTx: string, timestamp: string}[]
+}
+
+export interface IDbProposalPasswordData {
+  hashedAuthPassword: string,
+  authPasswordSalt: string,
+}
+
+export async function getProposalFromDb(mySql: ServerlessMysql, proposalId: string) {
+  const sqlRows: IProposalSqlRow[] = await mySql.query(
+    `SELECT proposal
+            , hashed_auth_password
+            , auth_password_salt
+            , version
+            , partial_tx
+            , signatures
+            , history
+            , updated_at 
+       FROM proposals
+      WHERE proposal = ?`,
+
+    [proposalId],
+  );
+
+  if (sqlRows.length === 0) {
+    return { dbProposal: null, dbProposalPasswordData: null };
+  }
+  if (sqlRows.length > 1) {
+    throw new LambdaError('Duplicate proposal ids found', ApiError.DuplicateProposalId);
```

Comment:
Add the proposal id on the message

Reply:
Good idea! Added on [cff1eab](https://github.com/HathorNetwork/hathor-atomic-swap-service/pull/3/commits/cff1eabff15094f8151bb6c6a03c42d8ba0c51e8)